### PR TITLE
DRIVERS-2543 Pull mongohouse image from ADL ECR repo

### PR DESCRIPTION
**Pull request by blink1073**
Pending https://github.com/10gen/mongohouse/pull/7602 we can remove the `test.yaml`

### CODE REVIEW

**Comment by blink1073** [comment]
Tested with https://spruce.mongodb.com/version/65485110e3c331c42e685c02/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: .evergreen/atlas_data_lake/run-mongohouse-image.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+#
+# This script launches a docker-based for mongohoused for testing.
```
Comment (NoahStapp) [inline .evergreen/atlas_data_lake/run-mongohouse-image.sh:3]: missing an "image" after "docker-based."

File: .evergreen/atlas_data_lake/build-mongohouse-local.sh
Comment (NoahStapp) [inline .evergreen/atlas_data_lake/build-mongohouse-local.sh:1]: Do we need this empty file?

File: .evergreen/atlas_data_lake/run-mongohouse-local.sh
Comment (NoahStapp) [inline .evergreen/atlas_data_lake/run-mongohouse-local.sh:1]: Do we need this empty file?

File: .evergreen/auth_aws/setup_secrets.py [hunk 2]
```diff
@@ -13,13 +13,17 @@
 
 AWS_ROLE_ARN = "arn:aws:iam::857654397073:role/drivers-test-secrets-role"
 
+AWS_ROLE_ARN = "arn:aws:iam::857654397073:role/drivers-test-secrets-role"
```
Comment (NoahStapp) [inline .evergreen/auth_aws/setup_secrets.py:16]: Duplicate definition of `AWS_ROLE_ARN`.

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/atlas_data_lake/build-mongohouse-local.sh
Comment (blink1073) [reply .evergreen/atlas_data_lake/build-mongohouse-local.sh:1]: They aren't empty, I made them executable so it shows up as a 0 line change.

File: .evergreen/atlas_data_lake/run-mongohouse-image.sh (hunk 1 above)
Comment (blink1073) [reply .evergreen/atlas_data_lake/run-mongohouse-image.sh:3]: Fixed

File: .evergreen/auth_aws/setup_secrets.py (hunk 2 above)
Comment (blink1073) [reply .evergreen/auth_aws/setup_secrets.py]: Fixed

File: .evergreen/atlas_data_lake/build-mongohouse-local.sh
Comment (NoahStapp) [reply .evergreen/atlas_data_lake/build-mongohouse-local.sh:1]: Got it, the Github UI does not make that clear.

**Review by NoahStapp — APPROVED** [review]
LGTM!

**Pull request merged**